HEN_HOUSE/egs++/egs_advanced_application.cpp

```diff
@@ -633,6 +633,7 @@ int EGS_AdvancedApplication::helpInit(EGS_Input *transportp, bool do_hatch) {
             the_xoptions->iedgfl = 2;
             relax.setOption(1,"eadl");
         }
+        delete transportp;
```

Comment:
`transportp` is allocated by `getInputItem` but wasn't freed.
https://github.com/nrc-cnrc/EGSnrc/blob/3e0db683e48a328b8fede9e7756e0deb693fbf9e/HEN_HOUSE/egs%2B%2B/egs_advanced_application.cpp#L435
https://github.com/nrc-cnrc/EGSnrc/blob/3e0db683e48a328b8fede9e7756e0deb693fbf9e/HEN_HOUSE/egs%2B%2B/egs_input.cpp#L245-L254